client/castApp/src/dbcb.c

```diff
@@ -71,12 +71,13 @@ static int pushEnv(caster_t *caster)
     }
 
     epicsMutexMustLock(caster->lock);
-    for (i = 0; !ret && i < caster->num_extra_envs; i++) {
-        const char *val = getenv(caster->extra_envs[i]);
+    for (i = 0; !ret && i < caster->extra_envs.count; i++) {
+        string_list_t *envptr = (string_list_t *)ellNth(&caster->extra_envs, i);
+        const char *val = getenv(envptr->item_str);
         if (val && val[0] != '\0')
-            ret = casterSendInfo(caster, 0, caster->extra_envs[i], val);
+            ret = casterSendInfo(caster, 0, envptr->item_str, val);
         if (ret)
-            casterMsg(caster, "Error sending env %s", caster->extra_envs[i]);
+            casterMsg(caster, "Error sending env %s", envptr->item_str);
```

Comment:
I think this iteration can more cleanly be done over the list as opposed to over the count and selecting the N-th element; this makes it an N^2 iteration over the list instead of a one-time pass.